src/tasks/task-joint-posVelAcc-bounds.cpp

```diff
@@ -176,226 +195,182 @@ namespace tsid
-              ">"<< m_dqMax[i]<<std::endl;
-            }
-            m_viabViol[i] =std::abs(dqa[i])-m_dqMax[i];
-
+          if(verbose)
+          {
+            std::cout << "State (q,dq) :("<<qa[i]<<","<<dqa[i]<<") of joint "
+            << i <<" is not viable because |dq|>dqMax : "<< std::abs(dqa[i]) <<
+            ">"<< m_dqMax[i]<<std::endl;
+          }
+          m_viabViol[i] =std::abs(dqa[i])-m_dqMax[i];
         }
         double dqMaxViab =   std::sqrt(std::max(0.0, 2*m_ddqMax[i]*(m_qMax[i]-qa[i])));
         if(dqa[i]>(dqMaxViab+eps))
         {
-            if(verbose)
-            {
-              std::cout << "State (q,dq,dqMaxViab) :("<<qa[i]<<","<<dqa[i]<<","<<
-              dqMaxViab<<") of joint "<< i <<" is not viable because dq>dqMaxViab : "
-              << dqa[i] <<">"<< dqMaxViab<<std::endl;
-            }
-            m_viabViol[i] =dqa[i]-dqMaxViab;
-
+          if(verbose)
+          {
+            std::cout << "State (q,dq,dqMaxViab) :("<<qa[i]<<","<<dqa[i]<<","<<
+            dqMaxViab<<") of joint "<< i <<" is not viable because dq>dqMaxViab : "
+            << dqa[i] <<">"<< dqMaxViab<<std::endl;
+          }
+          m_viabViol[i] =dqa[i]-dqMaxViab;
         }
         double dqMinViab =  -std::sqrt(std::max(0.0,2*m_ddqMax[i]*(qa[i]-m_qMin[i])));
         if(dqa[i]<(dqMinViab+eps))
         {
-            if(verbose)
-            {
-              std::cout << "State (q,dq,dqMinViab) :("<<qa[i]<<","<<dqa[i]<<","<<
-              dqMinViab<<") of joint "<< i <<" is not viable because dq<dqMinViab : "
-              << dqa[i] <<"<"<< dqMinViab<<std::endl;
-            }
-            m_viabViol[i]=dqMinViab-dqa[i];
+          if(verbose)
+          {
+            std::cout << "State (q,dq,dqMinViab) :("<<qa[i]<<","<<dqa[i]<<","<<
+            dqMinViab<<") of joint "<< i <<" is not viable because dq<dqMinViab : "
+            << dqa[i] <<"<"<< dqMinViab<<std::endl;
+          }
+          m_viabViol[i]=dqMinViab-dqa[i];
         }
       }
-
-      // if((m_viabViol == Vector::Zero(m_na))&&verbose==true)
-      // {
-      //   // std::cout << "State (q,dq) :("<<qa.transpose()<<","<<dqa.transpose()<<"is viable"<<std::endl;
-      //   std::cout << "State is viable"<<std::endl;
-      // }
     }
 
-    void TaskJointPosVelAccBounds::computeVelLimits(const Vector& q, bool verbose)
-     {
-        Vector qa = q.tail(m_na);
-        for(int i = 0; i<m_na; i++)
-        {
-        if(verbose)
+    void TaskJointPosVelAccBounds::computeAccLimitsFromPosLimits(const Vector&qa,
+                                                                  const Vector& dqa,
+                                                                  bool verbose)
+    {
+      m_ddqMax_q3 = m_two_dt_sq*(m_qMax-qa-m_dt*dqa);
+      m_ddqMin_q3 = m_two_dt_sq*(m_qMin-qa-m_dt*dqa);
+      m_ddqMax_q2 = Vector::Zero(m_na);
+      m_ddqMin_q2 = Vector::Zero(m_na);
```

Comment:
Have you verified whether this does not allocate dynamic memory? maybe using the method setZero would be better